GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -529,12 +532,25 @@ public FileSystemTaskResult AddMissingModifiedFiles()
             {
                 if (this.modifiedFilesInvalid)
                 {
-                    FileSystemTaskResult result = this.indexParser.AddMissingModifiedFiles(this.context.Tracer, this.indexFileStream);
+                    Dictionary<string, PlaceholderListDatabase.PlaceholderData> placeholderDataByPath;
+                    List<PlaceholderListDatabase.PlaceholderData> placeholdersToKeep;
+                    this.placeholderList.GetAllEntries(out placeholderDataByPath, folderPlaceholders: out placeholdersToKeep);

```

Comment:
I'm not quite following the changes here in this chunk of code. 

First question is, what is `GetAllEntries` returning? Are `placeholderDataByPath` and `placeholdersToKeep` duplicate data, just stored in different data structures? Are they mutually exclusive (and what's the division)? Are they separate but possibly overlapping sets?

Reply:
I think some better naming would help.

The flow in here is:

- `placeholderDataByPath` -> Populated with file placeholders by `GetAllEntries`.  This should be renamed to something like `placeholderFilesByPath`
- `placeholdersToKeep` -> Populated by folder placeholders by `GetAllEntries`.

As `AddMissingModifiedFiles` is walking the index, it will move entries from `placeholderDataByPath` to `placeholdersToKeep` if the files still exist in the index and have their skip-worktree bit turned on.

Then, when `AddMissingModifiedFiles` completes any items still left in `placeholderDataByPath` were either not in the index or they did not have their skip-worktree bit turned on.  Those entries should be added to ModifiedPaths.dat and removed from the placeholder list.

At the end of the method `WriteAllEntriesAndFlush` is called with `placeholdersToKeep` because it now contains:
- All folder placeholders
- All file placeholders that are still in the index with their skip-worktree bit turned on